eth/protocols/eth/handlers.go

```diff
@@ -121,15 +121,22 @@ func AnswerGetBlockHeadersQuery(db kv.Tx, query *GetBlockHeadersPacket, blockRea
 			}
 		case query.Reverse:
 			// Number based traversal towards the genesis block
-			if query.Origin.Number >= query.Skip+1 {
-				query.Origin.Number -= query.Skip + 1
-			} else {
+			current := query.Origin.Number
+			ancestor := current - (query.Skip + 1)
```

Comment:
When using minus - can also get uint underflow

Reply:
Gets taken care of by the if statement below